### ISSUE TITLE
Not possible to use skinnywar in Vaadin14

### ISSUE THREAD

**Issue by guttormvik** [vaadin/platform#1230]
### TLDR

With a large ear project with multiple war/jars, it is desirable to use skinnywar to keep the size down.
This does not work with Vaadin14

### Our existing Vaadin8 app

In our Vaadin8 based application we have 10 wars, 5 jars + one "widgetset" war.
Those wars that use vaadin depend on the "widgetset" war, and everything works fine.

We have also turned on "skinnywar" in the ear build, to place all shared jars in ear\lib and keep the overall size down.
With skinnywar on, or ear is 100M.
Without skinnywar, our ear is 180M. 
So skinnywar gives a significant saving that we'd like to keep when moving to Vaadin14.

### Trying the same in Vaadin14

So far in my Vaadin14 experiments it seems it is not possible to use skinnywar.
With Skinnywar on, and 2 war/jars depending on Vaadin, all vaadin jars end up in ear\lib, and they are unable to use the classloader to
1) access the frontend stuff in the war
2) Scan for @Route

As soon as I remove skinnywar, my tester ear project works.

Note: This issue is perhaps not present in all application servers. It depends how their classloader is implemented. We use wildfly 18, and there, if jar A depends on jar B, A's classloader can see B's classes, but B's classloader can't see A's.

In addition, it also doesn't seem possible in Vaadin14 to create a shared "widgetset". So far it seems all Vaadin-based wars must build their own copy of the client-side .

This means that for each Vaadin-based war I'll get an overhead of about:
* 3MB of frontend stuff
* 14MB of jars

( Based on my trivial starter project and Vaadin 14.2.0.alpha7 )

If I convert all 10 wars in my ear to vaadin, that means an overhead of 170M!
Adding in the 80MB of shared classes from the rest of our app, it means our 100MB Vaadi8 app becomes a 350MB Vaadin14 app!

### Possible solution?

With my simplistic understanding of Vaadin's internal workings, this could pehaps be solved if each war had an "entrypoint" class (MyServlet extends VaadinServlet perhaps?) and Vaadin used its classloader. This would then see everything in the war, even if the rest of vaadin's classes were in ear\lib.

With something like this, we would get rid of 14MB of the per-war overhead.
We would still have the 3MB of frontend stuff per war, but size-wise this would be a huge improvement.

**Comment by juhopiirainen** [comment]
Thanks for the issue @guttormvik . I move this to the Flow team for investigations.
